ocaml/xapi/xapi_pool_helpers.mli

```diff
@@ -32,7 +32,11 @@ val ha_disable_in_progress : __context:Context.t -> bool
 
 val ha_enable_in_progress : __context:Context.t -> bool
 
-val pool_secret_rotation_pending : __context:Context.t -> bool
+val assert_pool_secret_rotation_not_pending : __context:Context.t -> unit
+
+val assert_designate_new_master_not_in_progress : __context:Context.t -> unit
+
+val assert_cluster_create_not_in_progress : __context:Context.t -> unit
```

Comment:
These are probably redundant, the allowed operations checking in with_pool_operation should call the allowed operation checker and reject things there.